common/lib/common-definitions/src/logger.ts

```diff
@@ -29,11 +29,32 @@ export interface ITaggedTelemetryPropertyType {
 
 /**
  * JSON-serializable properties, which will be logged with telemetry.
+ * @deprecated interface will be repurposed to support flat arrays and objects.
  */
 export interface ITelemetryProperties {
     [index: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
 }
 
+/**
+ * Set of properties defined for the base logger to use only primitive types.
+ */
+// eslint-disable-next-line @rushstack/no-new-null
+export type TelemetryBaseEventPropertyType = string | number | boolean | null | undefined;
```

Comment:
Let's also update `ITelemetryBaseEvent` to extend this.  It will be a small breaking change they'll have to deal with during the integrate, but should be trivial.  And then we'll need an actual Breaking Change note (not just Upcoming).

The way I'm thinking about it is anything with "Base" in the name is part of our external API contracts that cross layers, e.g. that the host will implement.  So in that sense I really like this change in name.